actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -26,7 +26,42 @@ module TagHelper
       PRE_CONTENT_STRINGS[:textarea]  = "\n"
       PRE_CONTENT_STRINGS["textarea"] = "\n"
 
+      class TagBuilder #:nodoc:
+        include TagHelper
 
+        VOID_ELEMENTS = %w(base br col embed hr img input keygen link meta
+        param source track wbr).to_set
+
+        VOID_ELEMENTS.merge(VOID_ELEMENTS.map(&:to_sym))
+
+        def initialize(view_context)
+          @view_context = view_context
+        end
+
+        private
+          def tag_string(name, content_or_options = nil, options_or_escape = nil, escape = true, &block)
```

Comment:
@dhh you didn't mind people being able to put content in a br element. Still prefer that even if we can remove a code branch as shown just above?